broadbean/deprecate.py

```diff
@@ -0,0 +1,114 @@
+import types
```

Comment:
do we want to make the deprecate module public? i think it's fine, but then let's restrict the public parts of it to just the DeprecationWarning and deprecate? perhaps using `__all__` as in qcodes or via prepending underscore to everything else

Reply:
Can't we use the deprecate decorator from qcodes since that is public api?